### ISSUE TITLE
Recorder on mobile imply a jerky wav output

### ISSUE THREAD

**Issue by pierreclr** [linto-ai/WebVoiceSDK#4]
### Description 
When using the Recorder on a mobile, the generated wav is damaged.  
Mic options : 
```
{
      frameSize: 4096,
      constraints: {
        echoCancellation: true,
        autoGainControl: true,
        noiseSuppression: true
      }
    }
```
Attached to this issue you'll find a zip (github does not accept audio files to be attached) containing 3 audio files : 
* desktop.wav recorded on a desktop with WebVoiceSDK.Recorder
* mobile.wav recorded on an iPhoneX with WebVoiceSDK.Recorder
* usermedia-mediarecorder.webm recorded with native MediaRecorder and `navigator.mediaDevices.getUserMedia({ audio: true })`

[Archive.zip](https://github.com/linto-ai/WebVoiceSDK/files/6365599/Archive.zip)

Do you have any ideas on a way to solve this problem ? 
Maybe once I'll be able to start the project I can add "Record mic" "Stop record mic" buttons to the test page and consequently to https://webvoicesdk.netlify.app/
That way we'll be able to test the recorder across many devices. 

Thank you 

EDIT: After further investigation output wave audio is cleaner if I disable other intensive task (gesture recognition) on my webapp. Anyway that means transcoding AudioContext stream to WAV will be affected when available ressources are low. Maybe we can add options in the Recorder constructor to allow developers to opt for a more Native recording system.

**Comment by damienlaine** [comment]
Ok, there's a lot of things here that we need to consider :
1st : When i listen to "mobile.wav" there's obviously something wrong with the sample rate beside the cracks and pops, the audio seems sped up. That suggests that the raw information from the mic is not not written correctly. I cannot investigate this further ATM.
2nd : The cracks and pops : Yeah, it's related to performances issues. I audaciously relied everything on the native javascript "events / listener" implementation... i wanted to see if it's doable this way. And more or less, it is, but it requires a bit of tweaking. I already have a big pipeline of events and streams that's optimized with Web Workers. Maybe an approach could be to use another worker to actually record the utterances triggered by wake-up-word and ended by VAD.
Btw, if you use the https://webvoicesdk.netlify.app/ demo it records a lot of things along with audio (MFCC computations, Emphased audio frames, downsampled audioframes...) and might also output some cracks, the simple utterance acquisition , however, have proven to work flawlessly on low-end Android smartphones and i don't rly feel concerned by those Safari/Iphone crappynesses.

As you suggest, you might change the recorder to hook directly to the mic, and start the acquisition with the native browser implementation (it will mostly work the same as putting the recorder in a separate worker). But note that, as far as i'm concerned, i want to record from the Downsampler node, at a fixed 16Khz / 16bit mono.

If the performance drop is not reflected to the audio when using native recorder, it might however drift the performance on your super fat webcam acquisition,

In a nutshell, and as i like french idioms, you can't  make omelettes whitout cracking eggs.

**Comment by pierreclr** [comment]
It's perfectly clear, I'm closing the issue
